### ISSUE TITLE
modernize jdk level and dependencies

### ISSUE THREAD

**Issue by sreuland** [stellar/java-stellar-sdk#496]
### What problem does your feature solve?
jdk is pinned to LTS 1.8 from 2014 version.
uses some library deps that were prior required due to gaps in older jdk's such as threeten for date/time and to some degree google guava and okhttpclient.

### What would you like to see?
upgrade the jdk to LTS 11 which came out  in 2018. 
evaluate using builtin java.net.HTTPClient and removing dep on okhttpclient
use jdk built-in `java.time.*` instead of threeten dep

**Comment by Ifropc** [comment]
FYI: removing deps on okhttpclient will break existing users who provide their own instance of okhttpclient in order to customize client (i.e. timeouts, etc)

**Comment by sreuland** [comment]
> FYI: removing deps on okhttpclient will break existing users who provide their own instance of okhttpclient in order to customize client (i.e. timeouts, etc)

yes, thanks for pointing out, would be breaking change for `org.stellar.sdk.Server` since it exposes the http client internals like that with constructors and getter/setters. I updated details on description to include that scope for consideration.

**Comment by overcat** [comment]
Can we consider removing the `guava` dependency? After upgrading to version 1.8+, most of the functionalities we use in `guava` have alternatives in the standard library, except for `ImmutableList`, but is it worth introducing it just for `ImmutableList`?

**Comment by overcat** [comment]
Currently, `java.net.HTTPClient` cannot be used on the Android platform([source](https://developer.android.com/studio/write/java11-default-support-table)). Of course, we can design the http client as an interface and provide services through `java.net.HTTPClient` in this SDK, and then use `OkHttpClient` to provide services on the Android platform through SPI.
